### ISSUE TITLE
Untested SIMD exception: 0xfffffffa with DXVK

### ISSUE THREAD

**Issue by xDoge26** [ptitSeb/box86#847]
Im trying to use DXVK with 32bit dx11 game (Touhou lunar night) , im not sure is this box86 bug as dxvk works fine with dx9 game

<details>

```
root@localhost:~# TU_DEBUG=noconform MESA_VK_WSI_DEBUG=sw DISPLAY=:1 box86 wine /sdcard/Documents/TouhouLunaNights/touhou_luna_nights.exe                               Box86 with Dynarec v0.3.1 127a48f1 built on Jul 26 2023 06:16:50                                                Box86 with Dynarec v0.3.1 127a48f1 built on Jul 26 2023 06:16:50                                                Dynarec for ARM64, with extension: ASIMD AES CRC32 PMULL PageSize:4096 Running on Kryo V2 Cortex-A73 with 8 Cores
Params database has 24 entries                          Box64 with Dynarec v0.2.3 0928f50b built on Jul 27 2023 06:26:29
Using default BOX64_LD_LIBRARY_PATH: ./:lib/:lib64/:x86_64/:bin64/:libs64/
Using default BOX64_PATH: ./:bin/                       Counted 32 Env var
Looking for /root/wine/bin/wineserver
Rename process to "wineserver"
Using native(wrapped) libdl.so.2
Using native(wrapped) libc.so.6
Using native(wrapped) ld-linux-x86-64.so.2
Using native(wrapped) libpthread.so.0
Using native(wrapped) libutil.so.1
Using native(wrapped) librt.so.1
002c:err:wineboot:process_run_key Error running cmd L"C:\\windows\\system32\\winemenubuilder.exe -a -r" (2).
0078:err:wineusb:usb_init Failed to initialize libusb: Other error
0074:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0074:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0074:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0074:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
Dynarec for ARM64, with extension: ASIMD AES CRC32 PMULL PageSize:4096 Running on Kryo V2 Cortex-A73 with 8 Cores
Params database has 24 entries
Box64 with Dynarec v0.2.3 0928f50b built on Jul 27 2023 06:26:29
BOX64: Wine64 detected, WINEPRELOADRESERVE="000400000-000438000"
Using default BOX64_LD_LIBRARY_PATH: ./:lib/:lib64/:x86_64/:bin64/:libs64/
Using default BOX64_PATH: ./:bin/
Counted 34 Env var
Looking for /root/wine/bin/wine64
argv[1]="C:\windows\system32\conhost.exe"
argv[2]="--unix"
argv[3]="--width"
argv[4]="56"
argv[5]="--height"
argv[6]="51"
argv[7]="--server"
argv[8]="0x10"
Rename process to "wine64"
Using native(wrapped) libdl.so.2
Using native(wrapped) libc.so.6
Using native(wrapped) ld-linux-x86-64.so.2
Using native(wrapped) libpthread.so.0
Using native(wrapped) libutil.so.1
Using native(wrapped) librt.so.1
Using emulated /root/wine/bin/../lib/wine/x86_64-unix/ntdll.so
Using emulated /usr/lib/x86_64-linux-gnu/libgcc_s.so.1
Using emulated /root/wine/lib/wine/x86_64-unix/libwine.so.1
Using emulated /root/wine/lib/wine/x86_64-unix/win32u.so
Using native(wrapped) libm.so.6
Using native(wrapped) libfreetype.so.6
Using native(wrapped) libfontconfig.so.1
steamfile:Z:\sdcard\Documents\TouhouLunaNights/steam_app
id.txt
Steam being initialised with appId 851100
Steam_Init

[1]+  Stopped                 TU_DEBUG=noconform MESA_VK_WSI_DEBUG=sw DISPLAY=:1 box86 wine /sdcard/Documents/TouhouLunaNights/touhou_luna_nights.exe
root@localhost:~# Steam - SteamAPI_RestartAppIfNecessary
Steam - Steamworks_InitCEGLibrary
Steam - SteamAPI_InitSafe
SUCCESS!
SUCCESS!
Setting breakpad minidump AppID = 851100
Steam_SetMinidumpSteamID:  Caching Steam ID:  7656119478
9371904 [API loaded no]
Steam - SteamApps()->BIsSubscribed()
Steam - SteamClient()->SetWarningMessageHook()
Steam - Steamworks_SelfCheck()
Steam - Steam_UserStats_Init()
Steam - Steam_RemoteStorage_Init()
Steam - Steam_Screenshots_Init()
Steam - Steam_Friends_Init()
Steam - Steam_UGC_Init()
Logged in steam user id 1124073472
Found 0 subscribed items
g_bSteamInitialised being set to trueSteam_Init
LoadSteamLib and _steamInit both ok
Using emulated /root/wine/lib/wine/x86_64-unix/winex11.so
Using native(wrapped) libXext.so.6
Using native(wrapped) libX11.so.6
Using native(wrapped) libxcb.so.1
Using native(wrapped) libXau.so.6
Using native(wrapped) libXdmcp.so.6
Using native(wrapped) libXinerama.so.1
Using native(wrapped) libXxf86vm.so.1
Using native(wrapped) libXrender.so.1
Using native(wrapped) libXrandr.so.2
Using native(wrapped) libXcomposite.so.1
Using native(wrapped) libXi.so.6
Using native(wrapped) libXcursor.so.1
0024:fixme:msg:pack_message msg 80 (WM_SETICON) not supported yet
Attempting to set gamepadcount to 12
0024:fixme:dwmapi:DwmEnableMMCSS (1) stub
info:  Game: touhou_luna_nights.exe
info:  DXVK: v2.2
info:  Vulkan: Found vkGetInstanceProcAddr in winevulkan.dll @ 0x6fa83b10
info:  Built-in extension providers:
info:    Win32 WSI
info:    OpenVR
info:    OpenXR
info:  OpenVR: could not open registry key, status 2
info:  OpenVR: Failed to locate module
info:  Enabled instance extensions:
info:    VK_KHR_get_surface_capabilities2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
info:  Turnip Adreno (TM) 610:
info:    Driver : turnip Mesa driver 23.2.99
info:    Memory Heap[0]:
info:      Size: 1822 MiB
info:      Flags: 0x1
info:      Memory Type[0]: Property Flags = 0x7
info:      Memory Type[1]: Property Flags = 0xf
info:      Memory Type[2]: Property Flags = 0xb
warn:  DXGI: Found monitors not associated with any adapter, using fallback
warn:  D3D11DeviceFeatures: External memory features not supported
info:  D3D11CoreCreateDevice: Maximum supported feature level: D3D_FEATURE_LEVEL_11_1
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_1
info:  Device properties:
info:    Device : Turnip Adreno (TM) 610
info:    Driver : turnip Mesa driver 23.2.99
info:  Enabled device extensions:
info:    VK_EXT_attachment_feedback_loop_layout
info:    VK_EXT_custom_border_color
info:    VK_EXT_depth_clip_enable
info:    VK_EXT_extended_dynamic_state3
info:    VK_EXT_graphics_pipeline_library
info:    VK_EXT_non_seamless_cube_map
info:    VK_EXT_robustness2
info:    VK_EXT_shader_module_identifier
info:    VK_EXT_shader_stencil_export
info:    VK_EXT_transform_feedback
info:    VK_EXT_vertex_attribute_divisor
info:    VK_KHR_pipeline_library
info:    VK_KHR_swapchain
info:  Device features:
info:    robustBufferAccess                     : 1
info:    fullDrawIndexUint32                    : 1
info:    imageCubeArray                         : 1
info:    independentBlend                       : 1
info:    geometryShader                         : 1
info:    tessellationShader                     : 1
info:    sampleRateShading                      : 1
info:    dualSrcBlend                           : 1
info:    logicOp                                : 1
info:    multiDrawIndirect                      : 1
info:    drawIndirectFirstInstance              : 1
info:    depthClamp                             : 1
info:    depthBiasClamp                         : 1
info:    fillModeNonSolid                       : 1
info:    depthBounds                            : 1
info:    multiViewport                          : 1
info:    samplerAnisotropy                      : 1
info:    textureCompressionBC                   : 1
info:    occlusionQueryPrecise                  : 1
info:    pipelineStatisticsQuery                : 1
info:    vertexPipelineStoresAndAtomics         : 1
info:    fragmentStoresAndAtomics               : 1
info:    shaderImageGatherExtended              : 1
info:    shaderClipDistance                     : 1
info:    shaderCullDistance                     : 1
info:    shaderFloat64                          : 0
info:    shaderInt64                            : 0
info:    variableMultisampleRate                : 1
info:    shaderResourceResidency                : 0
info:    shaderResourceMinLod                   : 0
info:    sparseBinding                          : 0
info:    sparseResidencyBuffer                  : 0
info:    sparseResidencyImage2D                 : 0
info:    sparseResidencyImage3D                 : 0
info:    sparseResidency2Samples                : 0
info:    sparseResidency4Samples                : 0
info:    sparseResidency8Samples                : 0
info:    sparseResidency16Samples               : 0
info:    sparseResidencyAliased                 : 0
info:  Vulkan 1.1
info:    shaderDrawParameters                   : 1
info:  Vulkan 1.2
info:    samplerMirrorClampToEdge               : 1
info:    drawIndirectCount                      : 1
info:    samplerFilterMinmax                    : 0
info:    hostQueryReset                         : 1
info:    timelineSemaphore                      : 1
info:    bufferDeviceAddress                    : 0
info:    shaderOutputViewportIndex              : 1
info:    shaderOutputLayer                      : 1
info:    vulkanMemoryModel                      : 1
info:  Vulkan 1.3
info:    robustImageAccess                      : 0
info:    pipelineCreationCacheControl           : 1
info:    shaderDemoteToHelperInvocation         : 1
info:    shaderZeroInitializeWorkgroupMemory    : 0
info:    synchronization2                       : 1
info:    dynamicRendering                       : 1
info:  VK_AMD_shader_fragment_mask
info:    extension supported                    : 0
info:  VK_EXT_attachment_feedback_loop_layout
info:    attachmentFeedbackLoopLayout           : 0
info:  VK_EXT_conservative_rasterization
info:    extension supported                    : 0
info:  VK_EXT_custom_border_color
info:    customBorderColors                     : 1
info:    customBorderColorWithoutFormat         : 1
info:  VK_EXT_depth_clip_enable
info:    depthClipEnable                        : 1
info:  VK_EXT_extended_dynamic_state3
info:    extDynamicState3AlphaToCoverageEnable  : 1
info:    extDynamicState3DepthClipEnable        : 1
info:    extDynamicState3RasterizationSamples   : 1
info:    extDynamicState3SampleMask             : 1
info:  VK_EXT_fragment_shader_interlock
info:    fragmentShaderSampleInterlock          : 0
info:    fragmentShaderPixelInterlock           : 0
info:  VK_EXT_full_screen_exclusive
info:    extension supported                    : 0
info:  VK_EXT_graphics_pipeline_library
info:    graphicsPipelineLibrary                : 1
info:  VK_EXT_memory_budget
info:    extension supported                    : 1
info:  VK_EXT_memory_priority
info:    memoryPriority                         : 0
info:  VK_EXT_non_seamless_cube_map
info:    nonSeamlessCubeMap                     : 0
info:  VK_EXT_robustness2
info:    robustBufferAccess2                    : 1
info:    robustImageAccess2                     : 1
info:    nullDescriptor                         : 1
info:  VK_EXT_shader_module_identifier
info:    shaderModuleIdentifier                 : 1
info:  VK_EXT_shader_stencil_export
info:    extension supported                    : 1
info:  VK_EXT_swapchain_colorspace
info:    extension supported                    : 0
info:  VK_EXT_hdr_metadata
info:    extension supported                    : 0
info:  VK_EXT_transform_feedback
info:    transformFeedback                      : 1
info:    geometryStreams                        : 1
info:  VK_EXT_vertex_attribute_divisor
info:    vertexAttributeInstanceRateDivisor     : 1
info:    vertexAttributeInstanceRateZeroDivisor : 1
info:  VK_KHR_external_memory_win32
info:    extension supported                    : 0
info:  VK_KHR_external_semaphore_win32
info:    extension supported                    : 0
info:  VK_NVX_binary_import
info:    extension supported                    : 0
info:  VK_NVX_image_view_handle
info:    extension supported                    : 0
info:  Queue families:
info:    Graphics : 0
info:    Transfer : 0
info:    Sparse   : n/a
warn:  DXVK: No state cache file found
info:  DXVK: Graphics pipeline libraries supported
DirectX11: Using hardware device
info:  Adapter LUID 0: 0:3f4
0024:fixme:system:QueryDisplayConfig (00000002 0158B874 01D90358 0158B8B0 01D903A8 00000000): semi-stub
err:   readMonitorEdidFromKey: Failed to get EDID reg key size
err:   DXGI: Failed to parse display metadata + colorimetry info, using blank.
info:  Presenter: Actual swap chain properties:
info:    Format:       VK_FORMAT_B8G8R8A8_UNORM
info:    Color space:  VK_COLOR_SPACE_SRGB_NONLINEAR_KHR
info:    Present mode: VK_PRESENT_MODE_IMMEDIATE_KHR
info:    Buffer size:  640x360
info:    Image count:  3
info:    Exclusive FS: 1
warn:  DXGI: MakeWindowAssociation: Ignoring flags
info:  DXVK: Using 8 compiler threads
0120:fixme:seh:fpe_handler untested SIMD exception: 0xfffffffa. Might not work correctly
wine: Unhandled exception 0xc00002b5 in thread 120 at address 40020BCB (thread 0120), starting debugger...
0128:fixme:seh:fpe_handler untested SIMD exception: 0xfffffffa. Might not work correctly
012c:fixme:seh:fpe_handler untested SIMD exception: 0xfffffffa. Might not work correctly
0124:fixme:seh:fpe_handler untested SIMD exception: 0xfffffffa. Might not work correctly
011c:fixme:seh:fpe_handler untested SIMD exception: 0xfffffffa. Might not work correctly
0130:fixme:seh:fpe_handler untested SIMD exception: 0xfffffffa. Might not work correctly
Box86 with Dynarec v0.3.1 127a48f1 built on Jul 26 2023 06:16:50
winedbg detected, not launching it!
0120:err:seh:start_debugger Couldn't start debugger L"winedbg --auto 32 300" (1359)
Read the Wine Developers Guide on how to set up winedbg or another debugger
0124:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
012c:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
0130:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
0128:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
0120:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
011c:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
root@localhost:~#
[1]+  Exit 41                 TU_DEBUG=noconform MESA_VK_WSI_DEBUG=sw DISPLAY=:1 box86 wine /sdcard/Documents/TouhouLunaNights/touhou_luna_nights.exe
```

</details>

**Comment by ptitSeb** [comment]
This looks like a SIGBUS is happening.
This might be box86 fault (vulkan is tricky in 32bits as some struct needs allignement), or the mesa drvier doing a SIGBUS itself (unalinged float/double access cause SIGBUS on 32bits ARM).
DXVK should generaly works with dx9 games.

Try to use `BOX86_SHOWSEGV=1 BOX86_SHOWBT=1` to have more log details on the issue.

**Comment by ptitSeb** [comment]
Yeah, I gpu driver on ARMHF tend to SEGBUS a lot. I have issues with Radeon RX550/MESA on my side also.

**Comment by Hacker1245** [comment]
I can confirm the same thing happens when testing Just Cause 2 on Turnip.

**Comment by Hacker1245** [comment]
> This problem is fixed using DXVK 0.96, 1.0 doesn't work anymore. But bad workaround

Wonder what broke newer versions?
